services/app_api/test/test_example.py

```diff
@@ -0,0 +1,127 @@
+import json
+import unittest
+from main import db
+from main.api.models import ExampleTable
+from test.base import BaseTestCase
+
+# Helper function to add a sample string to the example_table
+
+
+def create_example_string(example_string):
+    example_table_element = ExampleTable(string_field=example_string)
+    db.session.add(example_table_element)
+    db.session.commit()
+    return example_table_element
+
+# TODO test get
+# TODO test post
+# TODO test put
+# TODO test get
+
+
+class TestExampleService(BaseTestCase):
+    def test_get_example(self):
+
+    def test_post_example(self):
+    def test_put_example(self):
+    def test_delete_example(self):
+
+
+class TestSpeciesService(BaseTestCase):
+    # Basic happy path, ensure the /species route behaves correctly
+    def test_get_species(self):
+        example_string_1 = create_example_string('example_string_1')
+        example_string_2 = create_example_string('example_string_2')
+        example_string_3 = create_example_string('example_string_3')
+        with self.client:
+            response = self.client.get('/api/example_endpoint')
+            data = json.loads(response.data.decode())
+            self.assertEqual(response.status_code, 200)
+            print(data)
+            # self.assertEqual(len(data['species']), 1)
+            # self.assertIn('Norway rat', data['species'][0]['name'])
+            # self.assertIn('Rattus norvegicus', data['species'][0]['latin'])
+            # self.assertIn('Brown or dark grey in colour.',
+            #               data['species'][0]['description'])
+            # self.assertIn('1880-01-01', data['species'][0]['introduction'])
+            # self.assertIn(category.name, data['species'][0]['category'])
+
+
+    # # Basic happy path, ensure the /species route behaves correctly for POST
+    # def test_species_creation(self):
```

Comment:
Same here

Reply:
removed the species stuff still need to work on tests